File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/Banned.java
package io.github.tanguygab.playerlistexpansion.filters;

import org.bukkit.OfflinePlayer;

import java.util.stream.Stream;

public class Banned extends Filter {

    @Override
    public Stream<OfflinePlayer> filter(Stream<OfflinePlayer> stream, OfflinePlayer viewer) {
        return stream.filter(OfflinePlayer::isBanned);
    }

}

File: src/main/java/io/github/tanguygab/playerlistexpansion/filters/Banned.kt
package io.github.tanguygab.playerlistexpansion.filters

import org.bukkit.OfflinePlayer

class Banned : Filter() {
    override fun filter(name: String?, viewer: OfflinePlayer?): Boolean {
        val player = getOffline(name)
        return player != null && player.isBanned
    }
}
